java/dagger/internal/codegen/BindingGraph.java

```diff
@@ -1120,6 +1118,23 @@ private boolean hasLocallyPresentOptionalBinding(ResolvedBindings resolvedBindin
               && !getLocalExplicitBindings(keyFactory.unwrapOptional(resolvedBindings.key()).get())
                   .isEmpty();
         }
+
+        /**
+         * Returns {@code true} if {@code resolvedBindings} contains a binding for which there is
+         * an explicit present binding in this component that was resolved previously.
+         */
+        boolean hasLocalContributionBinding(ResolvedBindings resolvedBindings) {
+          return resolvedBindings
+                .contributionBindings()
+                .stream()
+                .anyMatch(contributionBinding ->
+                    !getLocalExplicitBindings(resolvedBindings.key()).isEmpty()
+                  && parentResolver.isPresent()
+                  && parentResolver.get().getPreviouslyResolvedBindings(
```

Comment:
The argument to hasLocalContributionBinding is the previously resolved bindings from the parent. So this should always be true. You can remove it.

Maybe each of these `hasLocal…` methods' parameters should be renamed to "previouslyResolvedBindings" or "parentBindings" to make that clear.